helpers/run-app.ts

```diff
@@ -26,63 +26,80 @@ const createProcess = (
 // eslint-disable-next-line max-params
 export async function runApp(
   appPath: string,
+  template: string,
   frontend: boolean,
   framework: TemplateFramework,
   port?: number,
   externalPort?: number,
 ): Promise<any> {
-  let backendAppProcess: ChildProcess;
-  let frontendAppProcess: ChildProcess | undefined;
-  const frontendPort = port || 3000;
-  let backendPort = externalPort || 8000;
-
-  // Callback to kill app processes
-  process.on("exit", () => {
-    console.log("Killing app processes...");
-    backendAppProcess.kill();
-    frontendAppProcess?.kill();
-  });
-
-  let backendCommand = "";
-  let backendArgs: string[];
-  if (framework === "fastapi") {
-    backendCommand = "poetry";
-    backendArgs = [
-      "run",
-      "uvicorn",
-      "main:app",
-      "--host=0.0.0.0",
-      "--port=" + backendPort,
-    ];
-  } else if (framework === "nextjs") {
-    backendCommand = "npm";
-    backendArgs = ["run", "dev"];
-    backendPort = frontendPort;
-  } else {
-    backendCommand = "npm";
-    backendArgs = ["run", "dev"];
-  }
-
-  if (frontend) {
+  if (template === "extractor") {
```

Comment:
these changes are hard to undertstand - how about using a dedicated test file for extractor?